jina/peapods/gateway.py

```diff
@@ -282,6 +282,16 @@ def api(mode):
 
             content['mode'] = ClientMode.from_string(mode)
 
+            if 'top_k' in content:
+                top_k_queryset = jina_pb2.QueryLang()
+                top_k_queryset.name = 'VectorSearchDriver'
+                top_k_queryset.priority = 1
+                top_k_queryset.parameters['top_k'] = content['top_k']
+                if 'queryset' not in content:
+                    content['queryset'] = top_k_queryset
```

Comment:
Not sure about the syntax, but the `.append` below hints, that it should be `= [top_k_queryset]`.

Reply:
inside the `_generate` the type is checked and it can be a single one, but yes it may be more coherent to do this